test/ConfigDistributor.spec.js

```diff
@@ -197,4 +155,126 @@ describe('A DistributeConfig class', () => {
 
   });
 
+  context('when actually interacting with the file system', () => {
```

Comment:
in general, I see much coupling between `ConfigDistributor` and the various filesystem primitives that it uses so I'd see it as _the_ bottom layer to integration test; unit testing of its functions doesn't go very far. It follows that its integration tests (in this `context()`) could start from the main entry point `distribute()` and check its effects on a real, testing filesystem.